lib/internal/util/inspect.js

```diff
@@ -680,7 +683,9 @@ function findTypedConstructor(value) {
 // value afterwards again.
 function formatValue(ctx, value, recurseTimes, typedArray) {
   // Primitive types cannot have properties.
-  if (typeof value !== 'object' && typeof value !== 'function') {
+  if (typeof value !== 'object' &&
+      typeof value !== 'function' &&
+      !isUndetectableObject(value)) {
```

Comment:
We could inline the undefined check here. That way it would be less branches for the primitives.

Reply:
I'd rather keep everything super tidy just because of how weird of a case this is, and i think v8 should be able to inline this itself (i'm not an expert on the js inlining part of the jit but this is a very simple function)